### PR TITLE
Annotate app.go

### PR DESCRIPTION
**Pull request by faddat**
Closes: #XXX

## Description

While building Craft Economy and trying to become more familiar with the cosmos SDK, and familiarize others, I ended up annotating app.go, and found it really helpful, so I figured that I might as well annotate Osmosis' app.go, too!

______

For contributor use:

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

**Comment by ethanfrey** [comment]
This is nice to explain to people how it works.

File: app/app.go [hunk 1]
```diff
@@ -377,6 +466,8 @@ func NewOsmosisApp(
 		gammtypes.ModuleName, incentivestypes.ModuleName, lockuptypes.ModuleName, claimtypes.ModuleName,
 		poolincentivestypes.ModuleName, superfluidtypes.ModuleName, bech32ibctypes.ModuleName, txfeestypes.ModuleName,
 	)
+
+	// Tell the app's module manager how to set the order of BeginBlockers, which are run at the end of every block.
```
Comment (ValarDragon) [inline app/app.go:470]:
```suggestion
	// Tell the app's module manager how to set the order of EndBlockers, which are run at the end of every block.
```

**Review by ValarDragon — APPROVED** [review]
LGTM once suggested comment change is in! ( I don't have perms to push to the branch)

Thanks for adding this, should help onboarding!

**Comment by faddat** [comment]
Onboarding is gonna be an ecosystem-wide team effort....

**Comment by ValarDragon** [comment]
Wait what, why did merging sunny's PR close this, I thought github auto-updates PRs to re-target main >:(

**Comment by ValarDragon** [comment]
I don't have perms to reopen off your fork

**Comment by faddat** [comment]
hey no worries-- I'll push to osmosis-labs in the future and resubmit :)

**Pull request closed without merging**